src/hdmf/container.py

```diff
@@ -34,6 +34,8 @@ def __repr__(self):
 
     @property
     def object_id(self):
+        if self.__object_id is None:
```

Comment:
Why does this ever evaluate to `True`? It seems that this should evaluate to `True` only if `None` gets passed into `__new__` (i.e. see line 21). Why/where does that happen?

Reply:
It gets set to None here:
https://github.com/hdmf-dev/hdmf/blob/860e2ca7572f4adfab7397ae7f17af74df115a7e/src/hdmf/build/map.py#L1646-L1661

when reading a file where object_id is not set.

Reply:
Sorry, ignore that. One second...

Reply:
Ah, okay. Could you add a comment that explains why that happens? Going forward, `__object_id` should never be none, unless it's an old file. It would be good to document that's what it's solving.

Reply:
Actually, no that's right. The builder's object id is set to None there. I believe that happens and breaks there before: 
https://github.com/hdmf-dev/hdmf/blob/860e2ca7572f4adfab7397ae7f17af74df115a7e/src/hdmf/build/map.py#L1197-L1198

Reply:
It fails on line 1660 in map.py because docval does not allow None for the second parameter of `builder.set_attribute` (I think `'default': None` is required for docval to accept None there.)

```python
    @docval({'name': 'name', 'type': str, 'doc': 'the name of the attribute'},
            {'name': 'value', 'type': None, 'doc': 'the attribute value'})
    def set_attribute(self, **kwargs):
```